### PR TITLE
fix: file content could not be found

### PR DESCRIPTION
**Pull request by cyrvan-b**
In case of a scan launched on a directory within a repository, the related file content was not found.

### CODE REVIEW

File: tests/functional/iac/test_iac_scan_diff.py [hunk 1]
```diff
@@ -87,6 +120,43 @@ def test_iac_scan_diff_new_vuln(tmp_path: Path) -> None:
...
+    result = run_ggshield_iac_scan(*args, cwd=inner_dir_path, expected_code=1)
+
+    # AND vulnerability of file1.tf shows as unchanged
+    # AND vulnerability of file2.tf shows as new
+    assert "0 incidents deleted" in result.stdout
+    assert "1 incident remaining" in result.stdout
+    assert "1 new incident detected" in result.stdout
+    # AND details about the new vulnerability are shown
+    assert "file2.tf" in result.stdout
+    assert "The incident was found between" not in result.stdout
```
Comment (GG-HH) [inline tests/functional/iac/test_iac_scan_diff.py:157]: Maybe we should test the presence of parts of the `_IAC_SINGLE_VULNERABILITY` content ? That would be more robust, in particular if we change the wording of the fallback message.

Comment (GG-HH) [reply tests/functional/iac/test_iac_scan_diff.py:157]: And same below

**Review by GG-HH — CHANGES_REQUESTED** [review]

File: ggshield/iac/output/iac_text_output_handler.py [hunk 2]
```diff
@@ -71,16 +71,27 @@ def group_incidents_by_filename(
 class IaCTextOutputHandler(IaCOutputHandler):
     nb_lines: ClassVar[int] = 3
 
+    def _get_source_path(self, scan: IaCPathScanCollection) -> Path:
```
[the remark this answers is not in the document]
Comment (cyrvan-b) [reply ggshield/iac/output/iac_text_output_handler.py]: After some tests it looks like their behavior is the same. Maybe calling `git rev-parse --show-toplevel` is less error prone as this is provided directly with git. I will propose a solution with that.

**Comment by GG-HH** [comment]
Sounds good to me but there is a conflict with the main branch.

**Review by GG-HH — APPROVED** [review]

**Pull request merged**